### PR TITLE
cpu/stm32l1: handle I2C bus errors

### PR DESCRIPTION
**Pull request by olegart**
STM32L1 I2C bus error handling and recovery, including NACK handling (which is not even an error as it may be normally used by some devices, e.g. SHT21 sensor in No Hold Mode).

### CODE REVIEW

File: cpu/stm32l1/periph/i2c.c [hunk 1]
```diff
@@ -353,12 +376,30 @@ void i2c_poweroff(i2c_t dev)
...
+
+static int _start(I2C_TypeDef *i2c, uint8_t address, uint8_t rw_flag)
 {
     /* wait for device to be ready */
     DEBUG("Wait for device to be ready\n");
 
-    while (i2c->SR2 & I2C_SR2_BUSY) {}
+    uint32_t time_now = xtimer_now_usec();
+    while (i2c->SR2 & I2C_SR2_BUSY) {
+        /* 100 ms timeout */
```
Comment (haukepetersen) [inline cpu/stm32l1/periph/i2c.c]: why 100ms? I could not find anything in the standard (though I did no look very closely...). So if this is simply a value that turned out to be a good practice, I think it makes sense to a) document it, and/or b) make it configurable (I guess through either `periph_conf.h` or `periph_cpu.h`).

**Comment by olegart** [comment]
Hello

There's no specific meaning in 100 ms value, it is kind of "big enough not to raise false alarms"; at the same time, there's no actual reason to make it smaller as bus timeout is not a common situation, and usually a sign of hardware failure, so smaller timeout wouldn't improve power consumption or system performance.

So let's make it as #define, but inside i2c.c file. If someone really needs to, it can be redefined, but 9999 people out of 10000 will never care, so no need to push it in their faces through periph_conf or something like that.

We are using I2C quite often, and with modular devices and a range of sensors, so things like trying to access non-existing address or incorrectly connected device happen every day. With these fixes it works perfectly well; the only unrecoverable state is SCL permanently tied down to ground.

**Comment by MrKevinWeiss** [comment]
Hey @olegart, we are doing a major rework of the i2c.  I think @aabadie did something for the stm i2c cpu.  Would you be interested in rebasing to the new_if_branch (if the problem is still there)?  Let me know what you want done with this PR.

**Comment by MrKevinWeiss** [comment]
@olegart If you didn't want to update this PR to the next API (including errno defined error codes) I can take that over and update i2c_2.c?
Please let me know what you would like.

**Comment by MrKevinWeiss** [comment]
Since i2c has been reworked I believe the problem has been solved in a different way.

May I close this PR?

**Comment by MrKevinWeiss** [comment]
Closing since NACK are now being handled.

**Pull request closed without merging**